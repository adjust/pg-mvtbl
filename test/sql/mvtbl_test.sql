CREATE TABLESPACE mvtbl_test_tblspace LOCATION '/tmp/tsttblsp';

CREATE TABLE test AS SELECT i as a, i as b , i as c FROM generate_series(1,1e6) i;
CREATE INDEX ON test(a);
CREATE INDEX ON test(a,b);
CREATE INDEX ON test(c);

SELECT pg_size_pretty(mvtbl('test','mvtbl_test_tblspace'));
SELECT pg_size_pretty(mvtbl('test','pg_default'));

SELECT pg_size_pretty(mvtbl('test','mvtbl_test_tblspace', 3, 3));
SELECT pg_size_pretty(mvtbl('test','pg_default', 3));

ALTER EXTENSION mvtbl UPDATE TO '0.0.2';
ALTER EXTENSION mvtbl UPDATE TO '0.0.3';

DROP TABLESPACE mvtbl_test_tblspace;