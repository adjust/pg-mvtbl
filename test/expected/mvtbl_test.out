CREATE TABLESPACE mvtbl_test_tblspace LOCATION '/tmp/tsttblsp';
CREATE TABLE test AS SELECT i as a, i as b , i as c FROM generate_series(1,1e6) i;
CREATE INDEX ON test(a);
CREATE INDEX ON test(a,b);
CREATE INDEX ON test(c);
SELECT pg_size_pretty(mvtbl('test','mvtbl_test_tblspace'));
NOTICE:  done moving test
NOTICE:  done moving test_a_b_idx
NOTICE:  done moving test_a_idx
NOTICE:  done moving test_c_idx
 pg_size_pretty 
----------------
 123 MB
(1 row)

SELECT pg_size_pretty(mvtbl('test','pg_default'));
NOTICE:  done moving test
NOTICE:  done moving test_a_b_idx
NOTICE:  done moving test_a_idx
NOTICE:  done moving test_c_idx
 pg_size_pretty 
----------------
 123 MB
(1 row)

DROP TABLESPACE mvtbl_test_tblspace;
